src/display_list.rs

```diff
@@ -425,27 +431,87 @@ fn do_the_layout<'a, 'b, T: Layout>(
     (layouted_arena, solved_widths.non_leaf_nodes_sorted_by_depth, WordCache(word_cache))
 }
 
+fn get_nodes_that_need_scroll_clip(arena: &Arena<LayoutRect>, parents: &Vec<(usize, NodeId)>) -> BTreeMap<NodeId, (TypedRect<f32, LayoutPixel>, TypedRect<f32, LayoutPixel>)> {
+    let mut nodes = BTreeMap::new();
+    for (_, parent) in parents {
```

Comment:
This can be theoretically improved:

1. Unioning the rectangles is heavier than just looping through the children and summing up their width / height / padding + margin.
2. Scroll nodes only need to be inserted if the parent doesn't have `overflow: hidden` activated
3. Overflow for X and Y needs to be tracked seperately (for `overflow-x` / `overflow-y` separation), so there you'd have to track in which direction the inner_rect is overflowing.

However, this would be a bit of a larger change, so for now it would be ok to just put a `// TODO` comment there and leave it at that.

Reply:
1. You think so? I was kinda sure I looked at the unioning function and it didn't really more than that ...
2. Yes, as mentioned in the PR, this was not done yet due to apparently broken hit-testing which apparently wasn't broken :D
3. Yes, this is already tracked.

I will make those changes as well I guess as you already fixed hittesting.